src/controllers/blog.controller.ts

```diff
@@ -0,0 +1,127 @@
+  res.redirect('auth/login')
+}
+
+export const showBlogGet = asyncHandler(async (req, res) => {
+  const userId = req.session.user?.id
+  if (!checkUserAuthentication(req, res)) {
+    return
+  }
+  const blogs = await blogService.getBlogsByUserId(Number(userId))
+  if (blogs.length === 0) {
+    return res.redirect('/blog/create')
+  }
+  res.render('blog/index', { blogs })
+})
+export const showCreateBlog = asyncHandler(async (req, res) => {
+  const userId = req.session.user?.id
+  if (!checkUserAuthentication(req, res)) {
+    return
+  }
+  res.render('blog/create')
+})
+
+export const createBlogPost = asyncHandler(async (req, res) => {
+  const userId = req.session.user?.id
+  if (!checkUserAuthentication(req, res)) {
+    return
+  }
+  const { title, description, content } = req.body
+
+  const files = req.files as {
+    [fieldname: string]: Express.Multer.File[]
+  }
+
+  const imageUrl = files.imageUrl ? files.imageUrl[0] : undefined
+  const additionalImages = files.additionalImages || []
+
+  const newBlog = await blogService.createBlog({
+    userId: Number(userId),
+    title,
+    description,
+    content,
+    imageUrl,
+    additionalImages
+  })
+
+  res.redirect('/blog')
+})
+
+export const showBlogDetail = asyncHandler(async (req, res) => {
+  const blogId = parseInt(req.params.id)
+  const blog = await blogService.getBlogById(blogId)
+
+  if (!blog) {
+    req.flash('error', req.t('blog.notFound'))
+    return res.redirect('/blog')
+  }
+  res.render('blog/detail', { blog })
+})
+
+export const getBlogUpdatePage = asyncHandler(async (req, res) => {
+  const userId = req.session.user?.id
+  if (!checkUserAuthentication(req, res)) {
+    return
+  }
+
+  const blogId = parseInt(req.params.id)
+  const blog = await blogService.getBlogById(blogId)
+  res.render('blog/update', { blog })
+})
+
+export const updateBlogPosts = asyncHandler(async (req, res) => {
+  const userId = req.session.user?.id
+  if (!checkUserAuthentication(req, res)) {
+    return
+  }
+  const { id, title, description, content, currentImageUrl, currentAdditionalImages } = req.body
+  const imageUrl = req.files && req.files.imageUrl ? req.files.imageUrl[0] : currentImageUrl
+
+  const additionalImages =
+    req.files && req.files.additionalImages ? req.files.additionalImages : JSON.parse(currentAdditionalImages || '[]')
+  const updatedBlog = await blogService.updateBlog(
+    {
+      userId: Number(userId),
+      title,
+      description,
+      content,
+      imageUrl,
+      additionalImages
+    },
+    Number(id)
+  )
+
+  res.redirect('/blog')
+})
+
+export const deleteBlog = asyncHandler(async (req, res) => {
+  const userId = req.session.user?.id
+  if (!checkUserAuthentication(req, res)) {
+    return
+  }
+  const blogId = parseInt(req.params.id)
+  const blog = await blogService.getBlogById(blogId)
+  if (!blog) {
+    req.flash('error', req.t('blog.notFound'))
+    return res.redirect('/blog')
+  }
```

Comment:
phần xử lý này dùng chung đc cho các action get detail, update, delele -> e move ra function not export để dùng chung nhé